tfplan/json_plan.go

```diff
@@ -59,111 +50,27 @@ type jsonResourceChange struct {
 }
 
 // isCreate returns true if the action on the resource is ["create"].
-func (c *jsonResourceChange) isCreate() bool {
+func (c *ResourceChange) isCreate() bool {
 	return len(c.Change.Actions) == 1 && c.Change.Actions[0] == "create"
 }
 
-type childModule struct {
-	Address      string         `json:"address"`
-	Resources    []jsonResource `json:"resources"`
-	ChildModules []childModule  `json:"child_modules"`
+// compatibility shim until ResourceChange is expected by all callers.
+func (c *ResourceChange) Kind() string {
+	return c.Type
 }
 
-// jsonResource represent single Terraform resource definition.
-type jsonResource struct {
-	Module       string                 `json:"module"`
-	Name         string                 `json:"name"`
-	Address      string                 `json:"address"`
-	Mode         string                 `json:"mode"`
-	Kind         string                 `json:"type"`
-	ProviderName string                 `json:"provider_name"`
-	Values       map[string]interface{} `json:"values"`
+// ComposeTF12Resources is a thin wrapper around ReadResourceChanges as a compatibility shim.
+func ComposeTF12Resources(data []byte, schemas map[string]*schema.Resource) ([]ResourceChange, error) {
```

Comment:
I forget- are we able to change this argument to be named `_`?

Reply:
signs point to yes; changed in daf0333d